examples/twr_aloha/control.c

```diff
@@ -0,0 +1,268 @@
+        return false;
+    }
+    /* on completion of a range request setup an event to keep listening,
+       if is anchor */
+    dpl_eventq_put(dpl_eventq_dflt_get(), &_slot_event);
+    return true;
+}
+
+/**
+ * @brief API for receive timeout callback.
+ *
+ * @param[in] inst  Pointer to struct uwb_dev.
+ * @param[in] cbs   Pointer to struct uwb_mac_interface.
+ *
+ * @return true on success
+ */
+static bool _rx_timeout_cb(struct uwb_dev *inst, struct uwb_mac_interface *cbs)
+{
+    struct uwb_rng_instance *rng = (struct uwb_rng_instance *)cbs->inst_ptr;
+
+    if (inst->role & UWB_ROLE_ANCHOR) {
+        /* Restart receiver */
+        uwb_phy_forcetrxoff(inst);
+        uwb_rng_listen(rng, 0xfffff, UWB_NONBLOCKING);
+    }
+    return true;
+}
+
+/**
+ * @brief In the example this function represents the event context
+ * processing of the received range request which has been offloaded from
+ * ISR context to an event queue.
+ */
+static void _slot_complete_cb(struct dpl_event *ev)
+{
+    assert(ev != NULL);
+
+    struct uwb_rng_instance *rng = (struct uwb_rng_instance *)
+        dpl_event_get_arg(ev);
+    struct uwb_dev *inst = rng->dev_inst;
+
+    if (inst->role & UWB_ROLE_ANCHOR) {
+        uwb_rng_listen(rng, 0xfffff, UWB_NONBLOCKING);
+    }
+}
+
+/**
+ * @brief An event callback to send range request every RANGE_REQUEST_T_US.
+ * On every request it will switch the used ranging algorithm.
+ */
+static void uwb_ev_cb(struct dpl_event *ev)
+{
+    struct uwb_rng_instance *rng = (struct uwb_rng_instance *)ev->arg;
+    struct uwb_dev *inst = rng->dev_inst;
+
+    if (inst->role & UWB_ROLE_ANCHOR) {
+        if (dpl_sem_get_count(&rng->sem) == 1) {
+            uwb_rng_listen(rng, 0xfffff, UWB_NONBLOCKING);
+        }
+    }
+    else {
+        int mode_v[8] = { 0 }, mode_i = 0, mode = -1;
+        static int last_used_mode = 0;
+        if (IS_USED(MODULE_UWB_CORE_TWR_SS)) {
+            mode_v[mode_i++] = UWB_DATA_CODE_SS_TWR;
+        }
+        if (IS_USED(MODULE_UWB_CORE_TWR_SS_ACK)) {
+            mode_v[mode_i++] = UWB_DATA_CODE_SS_TWR_ACK;
+        }
+        if (IS_USED(MODULE_UWB_CORE_TWR_SS_EXT)) {
+            mode_v[mode_i++] = UWB_DATA_CODE_SS_TWR_EXT;
+        }
+        if (IS_USED(MODULE_UWB_CORE_TWR_DS)) {
+            mode_v[mode_i++] = UWB_DATA_CODE_DS_TWR;
+        }
+        if (IS_USED(MODULE_UWB_CORE_TWR_DS_EXT)) {
+            mode_v[mode_i++] = UWB_DATA_CODE_DS_TWR_EXT;
+        }
+        if (++last_used_mode >= mode_i) {
+            last_used_mode = 0;
+        }
+        mode = mode_v[last_used_mode];
+#ifdef UWB_TWR_ALGORITHM_ONLY_ONE
+        mode = UWB_TWR_ALGORITHM_ONLY_ONE;
+#endif
+        if (mode > 0) {
+            uwb_rng_request(rng, ANCHOR_ADDRESS, mode);
+        }
+    }
+
+    /* reset the callback if ranging is enabled */
+    if (_ranging_enabled_flag) {
+        dpl_callout_reset(&_rng_req_callout, RANGE_REQUEST_T_US);
+    }
+}
+
+void init_ranging(void *arg)
```

Comment:
Since is not used and this function is declared in the application, you could get rid of it:
```suggestion
void init_ranging(void)
```